sport/app/football/feed/Competitions.scala

```diff
@@ -101,6 +101,9 @@ trait Competitions extends implicits.Football {
 
   def matches: Seq[FootballMatch] = competitions.flatMap(_.matches)
 
+  def isAMatchInProgress(matches: Seq[FootballMatch], clock: Clock): Boolean =
+    matches.exists(game => game.isLive || game.date.minusMinutes(5).toLocalDate.isBefore(LocalDate.now(clock)))
+
```

Comment:
Here I think all we need to find out is the check if there's any match that is live right now. So using `exists` when the first live match is found the iteration is stopped as we don't care about the rest of list.

Also, the match date type was a `ZonedDateTime` and we wanted to compare it with the current `LocalDate`. I first subtracted 5 minutes from the `ZonedDateTime` and then converted it to `LocalDate`

Reply:
Careful about using `LocalDate`, because that always begs the question "local to where?"

Frontend deliberately sets the timezone to Sydney so that developers don't get tripped up by assuming the server's timezone will be UTC (or London). In this case it doesn't matter which timezone we use, as long as we are timezone-aware, since we're comparing two times. I think `isBefore` will take any timezone aware datetime type.

As a rule of thumb, best to always use UTC for server date calculations. The only cases where we need to think about timezones is when we're displaying times to our end-users. In this case, rather than converting to a `LocalDate`, I'd consider converting it to the UTC timezone before the comparison (if the conversion is even required).

Reply:
I think I didn't know well about what the localDate is. Thanks for pointing it out. 

Please correct me if I'm wrong, what I now understand is that `game.date` is `ZonedDateTime` which means it is showing a datetime and it has zone data. 

For example, if the zone is in `Asia/Calcutta` it could have this value `2022-02-03T15:35:46.816+05:30[Asia/Calcutta]` 

If I convert it to UTC `game.date.withZoneSameInstant(ZoneOffset.UTC)` it will then show the datetime on utc with this value `2022-02-03T10:05:46.875Z`

From what you said, now I can convert the current time to utc as well and the compare them both like the following:

```
val gameUtc = game.date.withZoneSameInstant(ZoneOffset.UTC)
val currentTimeUtc = ZonedDateTime.now.withZoneSameInstant(ZoneOffset.UTC)

// this checks if current time is between 5 minutes before the game start time and 5 minutes after the game start time
gameUtc.minusMinutes(5).isBefore(currentTimeUtc) && gameUtc.plusMinutes(5).isAfter(currentTimeUtc)
```